fortinet-fortiproxy/operations.py

```diff
@@ -0,0 +1,399 @@
+        except requests.exceptions.ReadTimeout:
+            raise ConnectorError(
+                'The server did not send any data in the allotted amount of time')
+        except requests.exceptions.ConnectionError:
+            raise ConnectorError('Invalid Credentials')
+        except Exception as err:
+            raise ConnectorError(str(err))
+
+
+def check_payload(payload):
+    updated_payload = {}
+    for key, value in payload.items():
+        if isinstance(value, dict):
+            nested = check_payload(value)
+            if len(nested.keys()) > 0:
+                updated_payload[key] = nested
+        elif value:
+            updated_payload[key] = value
+    return updated_payload
+
+
+def create_firewall_policy(config, params):
+    fp = FortiProxy(config)
+    endpoint = 'cmdb/firewall/policy'
+    query_parameter = {
+        'vdom': params.pop('vdom', ''),
+        'action': params.pop('action', ''),
+        'nkey': params.pop('nkey', '')
+    }
+    params.update({'type': Policy_Type.get(params.get('type'))}) if params.get('type') else ''
+    params.update({'action': params.get('action').lower()}) if params.get('action') else ''
+    custom_attributes = params.pop('custom_attributes', '')
+    if custom_attributes:
+        params.update(custom_attributes)
+    query_parameter = {k: v for k, v in query_parameter.items() if v is not None and v != ''}
+    data = check_payload(params)
+    response = fp.make_rest_call(endpoint, 'POST', params=query_parameter, data=json.dumps(data))
+    return response
+
+
+def get_firewall_policy(config, params):
+    fp = FortiProxy(config)
+    endpoint = 'cmdb/firewall/policy'
+    params.update({'action': params.get('action').lower()}) if params.get('action') else ''
+    query_parameter = {k: v for k, v in params.items() if v is not None and v != ''}
+    response = fp.make_rest_call(endpoint, 'GET', params=query_parameter)
+    return response
+
+
+def get_firewall_policy_details(config, params):
+    fp = FortiProxy(config)
+    endpoint = 'cmdb/firewall/policy/{0}'.format(params.pop('policyid'))
+    params.update({'action': params.get('action').lower()}) if params.get('action') else ''
+    query_parameter = {k: v for k, v in params.items() if v is not None and v != ''}
+    response = fp.make_rest_call(endpoint, 'GET', params=query_parameter)
+    return response
+
+
+def update_firewall_policy(config, params):
+    fp = FortiProxy(config)
+    endpoint = 'cmdb/firewall/policy/{0}'.format(params.get('policyid'))
+    query_parameter = {
+        'vdom': params.pop('vdom', ''),
+        'action': params.pop('action', ''),
+        'before': params.pop('before', ''),
+        'after': params.pop('after', '')
+    }
+    params.update({'type': Policy_Type.get(params.get('type'))}) if params.get('type') else ''
+    params.update({'action': params.get('action').lower()}) if params.get('action') else ''
```

Comment:
the if condition will always be false here as you are doing params.pop('action', '') above this